site/gatsby-site/src/graphql/variants.js

```diff
@@ -0,0 +1,85 @@
+import gql from 'graphql-tag';
+
+export const FIND_VARIANTS = gql`
+  query FindVariants($query: ReportQueryInput) {
+    reports(query: $query, limit: 999) {
+      report_number
+      title
+      date_published
+      tags
+      text_inputs
+      text_outputs
+    }
+  }
+`;
+
+export const FIND_INCIDENT_VARIANTS = gql`
+  query FindIncidentVariants($incident_id: Int!) {
+    incident(query: { incident_id: $incident_id }) {
+      incident_id
+      reports {
+        report_number
+        title
+        date_published
+        tags
+        text_inputs
+        text_outputs
+      }
+    }
+  }
+`;
+
+export const FIND_VARIANT = gql`
+  query FindVariant($query: ReportQueryInput) {
+    report(query: $query) {
+      report_number
+      title
+      date_published
+      tags
+      text_inputs
+      text_outputs
+    }
+  }
+`;
+
+export const CREATE_VARIANT = gql`
+  mutation CreateVariant($input: CreateVariantInput!) {
+    createVariant(input: $input) {
+      incident_id
+      report_number
+    }
+  }
+`;
+
+export const UPDATE_VARIANT = gql`
+  mutation UpdateVariant($query: ReportQueryInput!, $set: ReportUpdateInput!) {
+    updateOneReport(query: $query, set: $set) {
+      url
```

Comment:
You don't need to list `text_inputs` here?

Reply:
There is no need to return the `text_inputs` and `text_outputs` fields as part of the UpdateVariant mutation. Both fields are part of the `ReportUpdateInput` object but not part of the returned mutation fields.

Reply:
I think it is  a good idea to query those fields anyway because it allows the Apollo client to update its cache (even if it is not being used at the moment)